pkg/trino/driver/driver.go

```diff
@@ -15,6 +16,14 @@ import (
 
 const DriverName string = "trino"
 
+type customTransport struct {
+	client *trinoClient.TrinoClient
+}
+
+func (t *customTransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
You can add a compile time assert in this package to check that `customTransport` implements `http.RoundTripper`:
```go
var _ http.RoundTripper = &customTransport{}
```

It's more of an annotation for developers to make it clear how it'll be used.